src/main/ApplicationUtils.cpp

```diff
@@ -425,4 +425,54 @@ catchup(Application::pointer app, CatchupConfiguration cc,
     catchupInfo = app->getJsonInfo();
     return synced ? 0 : 3;
 }
+
+int
+publish(Application::pointer app)
+{
+    if (!checkInitialized(app))
+    {
+        return 1;
+    }
+
+    // set known cursors before starting maintenance job
+    ExternalQueue ps(*app);
+    ps.setInitialCursors(app->getConfig().KNOWN_CURSORS);
```

Comment:
it would be cleaner to just set the right config elements so that `app->start()` does what you want (in this case, I think you just need to set `STANDALONE` to true)

Reply:
`STANDALONE` is set before - in CommandLine.cpp:runPublish function, by calling to `Config::setNoListen`.

Maintenance job is normally started in `ApplicationImpl::start`, which we don't use in command-line `catchup` and `publish`. I think we could move calling maintenance to the end of catchup/publish process, or just ignore that and postpone the maintenance itself to normal stellar-core runs (but we can't be sure there will be no instances of stellar-core used only for command-line catchup and publish).

Reply:
You didn't understand my question: why not calling `start` if standalone is already set?